### PR TITLE
Add describe response v3

### PR DESCRIPTION
**Pull request by jsancio**

### CODE REVIEW

**Comment by jsancio** [comment]
@cruhland Can you review this?

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageDescribeSpec.scala [hunk 1]
```diff
@@ -149,7 +168,7 @@ final class PackageDescribeSpec
       )
     )
 
-    val Right(actualContent) = parse(response.contentString)
```
Comment (cruhland) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageDescribeSpec.scala]: Why did this change?

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageDescribeSpec.scala [hunk 2]
```diff
@@ -162,7 +181,7 @@ final class PackageDescribeSpec
       rpc.v1.model.DescribeRequest("helloworld", version)
     )
     assertResult(Status.Ok)(response.status)
-    val Right(packageInfo) = parse(response.contentString)
```
Comment (cruhland) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageDescribeSpec.scala]: Why did this change?

File: docs/schemas/cosmos/rpc/v3/model/DescribeResponse-schema.json [hunk 3]
```diff
@@ -0,0 +1,620 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "additionalProperties": false,
+    "definitions": {
+        "base64String": {
```
Comment (cruhland) [inline docs/schemas/cosmos/rpc/v3/model/DescribeResponse-schema.json]: Is there any way to put these definitions in another file and refer to them by name? Even though this file won't ever change, it's still hard to tell if this schema matches the "official" package schema.

File: docs/endpoints/package-describe.raml [hunk 4]
```diff
@@ -5,7 +5,7 @@ post:
     start the service, and command line extensions that are included with the package.
   headers:
     Accept:
-      enum: [application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v2]
+      enum: [application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v3]
```
Comment (cruhland) [inline docs/endpoints/package-describe.raml]: Should we include all supported versions?

File: docs/endpoints/package-describe.raml [hunk 5]
```diff
@@ -15,16 +15,17 @@ post:
   responses:
     200:
       body:
-        application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v2:
-          type: !include ../schemas/cosmos/rpc/v2/model/DescribeResponse-schema.json
+        application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v3:
```
Comment (cruhland) [inline docs/endpoints/package-describe.raml]: Should we include all supported versions?

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/rpc/v2/circe/MediaTypedEncoders.scala [hunk 6]
```diff
@@ -11,20 +11,30 @@ import com.twitter.util.Try
...
+    DispatchingMediaTypedEncoder(
+      Set(
+        MediaTypedEncoder(
+          encoder = rpc.v3.model.DescribeResponse.encoder.contramap(
+            rpc.v3.model.DescribeResponse.apply
+          ),
+          mediaType = MediaTypes.V3DescribeResponse
+        ),
+        MediaTypedEncoder(
+          encoder = rpc.v2.circe.Encoders.encodeV2DescribeResponse.contramap {
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/rpc/v2/circe/MediaTypedEncoders.scala]: Would this be a good opportunity to move the v1 and v2 encoders to the companion objects?

File: docs/schemas/cosmos/rpc/v3/model/DescribeResponse-schema.json (hunk 3 above)
Comment (jsancio) [inline docs/schemas/cosmos/rpc/v3/model/DescribeResponse-schema.json]: Yeah, but let's do that in a different PR. I think that you can reference schemas using URLs but we need to investigate that.

File: docs/endpoints/package-describe.raml (hunk 4 above)
Comment (jsancio) [inline docs/endpoints/package-describe.raml]: Part of my only wants to document the latest version. If they want an older version then they should look at an older Cosmos. What do you think?

File: docs/endpoints/package-describe.raml (hunk 5 above)
Comment (jsancio) [inline docs/endpoints/package-describe.raml]: See my other comment.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/rpc/v2/circe/MediaTypedEncoders.scala (hunk 6 above)
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/rpc/v2/circe/MediaTypedEncoders.scala]: Let me do that in a future PR when I have more time.

**Comment by jsancio** [comment]
@cruhland Thanks for the review. I made additional changes to remove support for describe response v1.

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageDescribeSpec.scala [hunk 7]
```diff
@@ -193,43 +167,61 @@ private object PackageDescribeSpec extends TableDrivenPropertyChecks {
...
+        "command" : {
+          "pip" : [
+          "dcos<1.0",
+          "git+https://github.com/mesosphere/dcos-helloworld.git#dcos-helloworld=0.1.0"
+          ]
+        }
+      }
+    }
+    """
+    // scalastyle:off line.size.limit
```
Comment (cruhland) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/PackageDescribeSpec.scala]: This should be `scalastyle:on`.

**Comment by cruhland** [comment]
@jsancio one minor typo, merge whenever!
